### ISSUE TITLE
Question about training data

### ISSUE THREAD

**Issue by hanzifan** [yaolinli/TimeChat-Online#17]
Thanks for your great work. I noticed that the video addresses in the annotations are lists of JPG images, but the downloaded data (such as llava-video-178k and video-chat-flash) are videos with MP4 extensions. Should I convert these videos into JPG files ourselves? If so, at what frame rate (fps) should we extract the frames?

**Comment by yaolinli** [comment]
Hi, we extracted frames at 1 fps (1 frame per second) from the original videos. 
By the way, we have also released the extracted frames at [yaolily/TimeChat-Online-139K](https://huggingface.co/datasets/yaolily/TimeChat-Online-139K), where you can download them directly.

**Comment by hanzifan** [comment]
> Hi, we extracted frames at 1 fps (1 frame per second) from the original videos. By the way, we have also released the extracted frames at [yaolily/TimeChat-Online-139K](https://huggingface.co/datasets/yaolily/TimeChat-Online-139K), where you can download them directly.

Thank you for your reply, but I noticed that TimeChat-Online uses not only online data for training, but also offline data such as llava-video-178k. However, the downloaded llava-video-178k data consists of MP4 files. Should I extract video frames from these files at 1 fps by myself?

**Comment by hanzifan** [comment]
One more question, how can I speed up the training. I use 8 A800 for training just on the TimeChat-Online-139K , it seems like that needs 1 day for training.

<img width="1842" height="335" alt="Image" src="https://github.com/user-attachments/assets/1ca08f06-db3b-442e-8616-a9237c0e33cd" />
And why does it print so many "hello"

**Comment by wyclike** [comment]
For the "hello" issue, it is a leftover debug print statement that was forgotten to be commented out. You can find it at [this line](https://github.com/yaolinli/TimeChat-Online/blob/main/train/transformers-replace-code/modeling_qwen2_5_vl.py#L1373) and comment it out.

**Comment by wyclike** [comment]
For the offline data, we recommend extracting video frames from these files at 1 fps.

**Comment by hanzifan** [comment]
> For the "hello" issue, it is a leftover debug print statement that was forgotten to be commented out. You can find it at [this line](https://github.com/yaolinli/TimeChat-Online/blob/main/train/transformers-replace-code/modeling_qwen2_5_vl.py#L1373) and comment it out.

Thanks for your reply. Now I have met a strange problem. The loss will be nan after 1st backward when I just use Time-Chat-Online-139k for training. And I just use a part of the Time-Chat-Online-139k data also meets this problem. 

<img width="1281" height="538" alt="Image" src="https://github.com/user-attachments/assets/ca6ba634-e292-4634-a34a-3a5a1b74608f" />
Here I set gradient_accumulation_steps into 1 for debug. Other settings are just follow the original code.

**Comment by hanzifan** [comment]
> > For the "hello" issue, it is a leftover debug print statement that was forgotten to be commented out. You can find it at [this line](https://github.com/yaolinli/TimeChat-Online/blob/main/train/transformers-replace-code/modeling_qwen2_5_vl.py#L1373) and comment it out.
> 
> Thanks for your reply. Now I have met a strange problem. The loss will be nan after 1st backward when I just use Time-Chat-Online-139k for training. And I just use a part of the Time-Chat-Online-139k data also meets this problem.
> 
> <img alt="Image" width="1281" height="538" src="https://private-user-images.githubusercontent.com/43206681/477392244-ca6ba634-e292-4634-a34a-3a5a1b74608f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUwNjcxMDgsIm5iZiI6MTc1NTA2NjgwOCwicGF0aCI6Ii80MzIwNjY4MS80NzczOTIyNDQtY2E2YmE2MzQtZTI5Mi00NjM0LWEzNGEtM2E1YTFiNzQ2MDhmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODEzVDA2MzMyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjOThjOTU5NmFjNDE0MmE3ZTc3M2IxY2NiZjAxZjAxNWEwNWJkNjE0ODI4YjNkM2FlYTg3ZjcwNWY5MjhhYmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tPv6c6CDSQXVUWakNctvz_kuaQuLIWwfE_XWh2QO0Po"> Here I set gradient_accumulation_steps into 1 for debug. Other settings are just follow the original code.

Sorry I just print the logits, also the loss is nan too.

**Comment by wyclike** [comment]
It's really strange. I will test it again then reply you.If you could provide any additional information or possible insights, that would be very helpful

**Comment by wyclike** [comment]
I couldn't reproduce your error when set the gradient_accumulation_steps into 1. This is my log picture.

I think maybe the torch version you use is unmatch? (For reference,my torch version is 2.4.0 and my torchvision version is 0.19.0)

<img width="3194" height="1196" alt="Image" src="https://github.com/user-attachments/assets/e6ecf394-551c-4fde-bd6c-228c2b7d4ed0" />

**Comment by hanzifan** [comment]
> I couldn't reproduce your error when set the gradient_accumulation_steps into 1. This is my log picture.
> 
> I think maybe the torch version you use is unmatch? (For reference,my torch version is 2.4.0 and my torchvision version is 0.19.0)
> 
> <img alt="Image" width="2000" height="1196" src="https://private-user-images.githubusercontent.com/120073874/477859958-e6ecf394-551c-4fde-bd6c-228c2b7d4ed0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxNjY0NDAsIm5iZiI6MTc1NTE2NjE0MCwicGF0aCI6Ii8xMjAwNzM4NzQvNDc3ODU5OTU4LWU2ZWNmMzk0LTU1MWMtNGZkZS1iZDZjLTIyOGMyYjdkNGVkMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxNFQxMDA5MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZDFkZTM1MzI5NDM3MTk3MDc0NGRlZmVkYmY4MjEyNWEwM2UyMGYxZWU0YmVmYjIyYzcwZTc5NGNhMmU5ZjkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8ifk-1mFpcaRqVTuqRHej9xfPEYfH-vqameXNMVPkAM">

Thanks for your advice, I change torch version to 2.4.0 and torchvision version to 0.19.0 then I solve this problem.

**Comment by SamorsetTuska** [comment]
> > Hi, we extracted frames at 1 fps (1 frame per second) from the original videos. By the way, we have also released the extracted frames at [yaolily/TimeChat-Online-139K](https://huggingface.co/datasets/yaolily/TimeChat-Online-139K), where you can download them directly.
> 
> Thank you for your reply, but I noticed that TimeChat-Online uses not only online data for training, but also offline data such as llava-video-178k. However, the downloaded llava-video-178k data consists of MP4 files. Should I extract video frames from these files at 1 fps by myself?

Hello,  have you already extracted the frames from the offline video data? I encountered some problems during the extraction process. For instance, "Tarsier2-Recap-585K/Video-10M/2487878.mp4" is a 10-second short video, but there is "tarsier2/Video-10M/frames/2_6/2487878.mp4/000247.jpg" in the annotations.
